common/python/generate_app.py

```diff
@@ -301,8 +301,8 @@ def generate_soft_blocks(self):
                              "soft_blocks.vhd")
         self.expand_template("addr_defines.vhd.jinja2", context, hdl_dir,
                              "addr_defines.vhd")
-        self.expand_template("top_defines.vhd.jinja2", context, hdl_dir,
-                             "top_defines.vhd")
+        self.expand_template("top_defines_gen.vhd.jinja2", context, hdl_dir,
```

Comment:
Can you also make sure this file gets generated for the testbenches. Maybe at the same time as the module.tcl? You may need to generate some dummy parameters for total_pos_bus_length